### ISSUE TITLE
Inclusion of "Building:part" or "buildingpart" tags in index

### ISSUE THREAD

**Issue by justonefix** [twain47/Nominatim#294]
Dear Nominatim Team,

I am trying to set-up a local Nominatim instance, in order to geocode OSM building rooms for indoor routing with OTP. However, I cannot get Nominatim to process any additional tags, apart from the predefined ones. I.e. "building" works, however "building:part" (as per Simple Indoor tagging) or "buildingpart" (IndoorOSM) tags cannot be parsed. I have added these to `default.style`, `partitionedtags.def`, `phrase_settings.php` whitelist and in `functions.sql`, I have added them in the same expression alongside the house & building tags.
Despite the above changes, the rooms are not included in the `place` or `placex` tables and are not displayed in the geocoder results.
Changing the node tag to `building` makes it searchable.
The OSM data used for the above is retrieved via JOSM from the main DB and exported to PBF for use with Nominatim, in case someone wants to try this. More specifically, I am testing with an LMU building tagged as per the IndoorOSM schema in Munich, Germany. Bounding box is: min lat/lon (48.1492717,11.5931533) max lat/lon (48.1501299,11.594979).
Thanks in advance for any support here!

Best Regards,
George Mouratidis

**Comment by lonvia** [comment]
The choice of tags is currently hard-coded in osm2pgsql. Have a look at the file `osm2pgsql/output-gazetteer.cpp` and there in particular in the function `place_tag_processor::process_tags()`. Note that the `building` tag is handled a bit special (uninteresting buildings without names or similar are thrown away), so you probably want to handle your tags similar to POI tags like [shop](https://github.com/openstreetmap/osm2pgsql/blob/master/output-gazetteer.cpp#L190). Just add the required tags there copy&paste style, recompile, reimport and it should work.

**Comment by justonefix** [comment]
Dear Ionvia,

Thank you very very much, I have added both `building:part` and `buildingpart` under mountainpass in the POI section as you advised and both the geocoder and OTP are now parsing rooms and other (named) building parts fine!
I was wondering, are the rest of the changes I described (apart from the `output-gazeteer.cpp` that you provided) necessary, or would indexing work fine with the original versions of `functions.sql` `phrase_settings.php` `default.style` and `partitionedtags.def`?
Just wanted to make sure for my documentation of the necessary changes for building part support.

Again, thanks, it was a tedious couple of hours of rebuilding the index, trying to figure out why the tags were being ignored before your answer!

Best Regards,
George
